src/unitxt/operators.py

```diff
@@ -179,6 +179,127 @@ def process(
         return flatten_dict(instance, parent_key=self.parent_key, sep=self.sep)
 
 
+class WholeInputFormatter(StreamInstanceOperator):
+    r"""Generates the whole input to the model, from constant strings that are given as args, and from values found in specified fields of the instance.
+
+    WholeInputFormatter expects its input instance to have been processed by a Template StreamInstanceOperator
+    (from templates.py), or any extension thereof, and hence to contain fields named "source" and "target", and to have
+    been processed by RenderInstruction StreamInstanceOperator (from renderer.py) and hence to contain a non None
+    field "instruction".
+    Demos are assumed to be included as well, in a field whose name is specified by argument 'demos_field'. That
+    field is assumed to contain a list of dicts, each containing keys: "source" and "target" of a single demo.
+    And finally, system_intro, and system_closing (an introduction and closing texts) are also assumed to be included
+    in the instance, in fields whose names are specified by arguments 'system_intro' and 'system_closing'.
+    The value in field "source" of the input instance describes (verbalizes) original values in the instance (as read
+    from the source dataset), in the context of the underlying task.
+    WholeInputFormatter makes this (verbalization of the) input into a whole-input-to-the-model, by combining it
+    with instruction, demos, system_intro, and system_closing, read from the input instance, through several
+    formatting arguments.
+
+    Args:
+        to_field (str): the name of the field into which the formatted model's input is to be stored
+        demos_field (str): the name of the field that contains the demos, being dicts with "source" and "target" keys
+        system_intro (str): the name of the field containing an introductory text, or signs
+        system_closing (str): the name of the field containing text or signs to append at the end of the whole-input-to-the-model
+        input_prefix: str = ""
+        output_prefix: str = ""
+        target_prefix: str = " "
+        instruction_prefix: str = ""
+        input_output_separator: str = "\n"
+        demo_separator: str = "\n\n"
+        add_instruction_at_start: bool = True
+        add_instruction_after_demos: bool = False
+        overall_output_size_limit: int = None
+    """
+
+    to_field: str
+    demos_field: str = None
+    system_intro: str = None
+    system_closing: str = None
+    input_prefix: str = ""
+    output_prefix: str = ""
+    target_prefix: str = " "  # only for compatibility with ICLFormat.  surely sure?
```

Comment:
@elronbandel , please note the exceptional one space here

Reply:
This is an error and indeed needs to be removed.

Reply:
OK. Now that we all saw that `ModelInputFormatter` can do anything `ICLFormat` could do thus far, in all cases and tests; even to imitate this error, I will remove `ICLFormat` altogether. At this moment, it only lives in tests, just to prove that the same targets it was expecting, also `ModelInputFormatter` achieves. And on this occasion, @yoavkatz , I will fix that said error.